### DISCUSSION TITLE
Portable Git - Portable Credentials?

### DISCUSSION THREAD

**Discussion by ccoenen** [git-for-windows/git#4663] [Q&A]
I am in a situation where students would have to work on shared PCs. Those shared PCs have a single shared user account and students work off of USB drives most of the time. Yes, this situation is not ideal, I am aware.

I am looking into ways that the students might _still_ be able to mostly use git without too much hassle. I would like to avoid getting their login information into the windows credential manager, because I am pretty sure that they will not usually remove it from there.

I was wondering if a portable version of git for windows could be configured to

- store credentials
- but, crucially, not store it in the host system's credential manager

**Comment by dscho** [comment]
@mjcheetham do you have any thoughts on this?

**Comment by ccoenen** [comment]
We have since created a (janky) workaround, where we include gitconfigs from the USB Thumbdrive letters, and if they are there, they override settings like user.name or user.email. Sadly, one can't use relative filenames, or we could use it to also set the location of a credential store. But that credential store is not encrypted anyway, so I wouldn't want to take this from PC to PC on a thumbdrive anyway.

We have also created a script to purge login info from the credential store.

It's not solving my problem, but at least it's solving part of my problem.
